### PR TITLE
[ADD] Starshipit as a delivery provider

### PR DESCRIPTION
**Pull request by vin-odoo**
Starshipit is the leading provider of integrated
and automated fulfilment solutions for online businesses integrating with the main carriers used in Australia and New Zealand, such as Australia Post, DHL, Aramex, CouriersPlease and more

Task ID # 3041982

### CODE REVIEW

File: content/applications/inventory_and_mrp/inventory/shipping_receiving/setup_configuration/starshipit_shipping.rst
Comment (Felicious) [inline content/applications/inventory_and_mrp/inventory/shipping_receiving/setup_configuration/starshipit_shipping.rst:62]: I wanted to help add some additional details about what a shipping rule does and how it's used, for beginners who don't really understand why this is necessary (:
```suggestion
Configure shipping rules to assign the correct shipping methods to orders based on specific
conditions.

To create a rule, go to :menuselection:`Settings --> Rules` and click :guilabel:`Add a new rule`.

While there are multiple ways to configure rules, it is recommended to set:

#. :guilabel:`Condition` to :guilabel:`Contains`
#. :guilabel:`Value` to the `product code`
#. :guilabel:`Action` to :guilabel:`Set Courier & Product Code`
```

File: content/applications/inventory_and_mrp/inventory/shipping_receiving/setup_configuration/starshipit_shipping.rst
Comment (Felicious) [inline content/applications/inventory_and_mrp/inventory/shipping_receiving/setup_configuration/starshipit_shipping.rst:86]: I couldn't find Starshipit module in the 17.0 or master runbot.

image

Am I searching for it properly?

File: content/applications/inventory_and_mrp/inventory/shipping_receiving/setup_configuration/starshipit_shipping.rst
Comment (Felicious) [inline content/applications/inventory_and_mrp/inventory/shipping_receiving/setup_configuration/starshipit_shipping.rst:192]:
```suggestion
This can be done by pressing the :guilabel:`Return` smart button on the intended delivery order.
```

File: content/applications/inventory_and_mrp/inventory/shipping_receiving/setup_configuration/starshipit_shipping.rst
Comment (Felicious) [inline content/applications/inventory_and_mrp/inventory/shipping_receiving/setup_configuration/starshipit_shipping.rst:178]: Slight rewrite and add blank spaces between paragraphs to separate them visually.
```suggestion
   Package weight in Odoo is calculated by adding the weights of the products plus the empty package
   saved in the database. Ensure the correct shipping option is selected, as the package weight is
   not automatically verified.

   Verify the destination address, as Starshipit checks it when the order is created.

   Finally, some couriers may require other information, such as an email address or phone number.
   Please ensure that all necessary information are set upon sending a shipping order.
```

**Review by Felicious — CHANGES_REQUESTED** [review]
Hi @vin-odoo !

This was an incredibly detailed and well-written document, with lots of helpful screenshots. It was easy to follow and will be incredibly helpful to users. I'd like to express how grateful I am for your hard work in producing new and helpful content for the Odoo community. 

As a reviewer, I'm supposed to test the Starshipit module and its functionality to approve this pull request (PR). Since I couldn't find the module in the `master` or `17.0` runbot, I can't approve this PR at this time. Where did you test the functionality of this new feature?

In my review, I made some rewording suggestions and added some more details on how to navigate to certain menus. You can accept/ignore them as you see fit. Again, thank you so much for writing this new document, and feel free to reach out to me if you have any questions about my comments, or if you'd like a second review!

Warm regards,
Felicia

File: content/applications/inventory_and_mrp/inventory/shipping_receiving/setup_configuration/starshipit_shipping.rst
Comment (vin-odoo) [reply content/applications/inventory_and_mrp/inventory/shipping_receiving/setup_configuration/starshipit_shipping.rst:86]: Hello! The module is on a pull request linked to this one so that both get released at the same time.
https://58697424-17-0-all.runbot114.odoo.com/web#id=21&cids=6&menu_id=1170&action=1482&model=delivery.carrier&view_type=form

Here you can find a runbot with the app installed.
To ease your test, I went and created a starshipit shipping method with our test API keys.
The AU Company, and the partner Deco Addict have been set up with addresses that are correct to be used for testing.
I recommend keeping sendle drop off as shipping service as it's easier for me to cancel orders afterward :)

Just let me know when you are done.
Thanks!

Comment (vin-odoo) [reply content/applications/inventory_and_mrp/inventory/shipping_receiving/setup_configuration/starshipit_shipping.rst:86]: Also if you need access to the starshipit website to check there too, let me know and I'll share the credentials with you on discord :smile:

File: content/applications/inventory_and_mrp/inventory/shipping_receiving/setup_configuration/starshipit_shipping.rst
Comment (Felicious) [inline content/applications/inventory_and_mrp/inventory/shipping_receiving/setup_configuration/starshipit_shipping.rst:67]: Adding an anchor so we can link to this section later, when we need to fill out the API key in Odoo

```suggestion

.. _inventory/shipping_receiving/star-api:

```

File: content/applications/inventory_and_mrp/inventory/shipping_receiving/setup_configuration/starshipit_shipping.rst
Comment (Felicious) [inline content/applications/inventory_and_mrp/inventory/shipping_receiving/setup_configuration/starshipit_shipping.rst:119]: Reordered the items in this bulleted list to match the order in which they appear on the form in Odoo
```suggestion
In the :guilabel:`Starshipit Configuration` tab, fill out these fields:

- :guilabel:`Starshipit API Key`: enter the :abbr:`API (Application Programming Interface)` key
  :ref:`obtained from Starshipit <inventory/shipping_receiving/star-api>`.
- :guilabel:`Starshipit Subscription Key`: enter the subscription key obtained from the same place
  as the :ref:`API key <inventory/shipping_receiving/star-api>`.
- :guilabel:`Origin Address`: Enter the address where products are shipped from. This field is
   crucial for calculating shipping rates and :ref:`generating shipping labels
   <inventory/shipping_receiving/star-label>`.
- :guilabel:`Default Package Type`: Set a default package type to include the weight of the empty
   package when automatically calculating shipping rates.

.. important::
   To set a default package type, the *Packages* feature **must** be enabled in
   :menuselection:`Inventory --> Configuration --> Settings`.
```

File: content/applications/inventory_and_mrp/inventory/shipping_receiving/setup_configuration/starshipit_shipping.rst
Comment (Felicious) [inline content/applications/inventory_and_mrp/inventory/shipping_receiving/setup_configuration/starshipit_shipping.rst:152]:
```suggestion

.. _inventory/shipping_receiving/star-label:

```

File: content/applications/inventory_and_mrp/inventory/shipping_receiving/setup_configuration/starshipit_shipping.rst
Comment (Felicious) [inline content/applications/inventory_and_mrp/inventory/shipping_receiving/setup_configuration/starshipit_shipping.rst:129]: Totally personal preference, but we can go into more detail describing the pop-up window
```suggestion
To load the newly configured shipping products, click the :guilabel:`Select a service linked to your
Starshipit account` link at the bottom of the :guilabel:`Starshipit Configuration` tab.

Doing so opens the :guilabel:`Choose Starshipit Shipping Service` pop-up window. In the
:guilabel:`Delivery Service` field, choose the desired shipping service for deliveries and returns
from the drop-down menu. Finally, click :guilabel:`Confirm`.

The chosen delivery service will populate in the :guilabel:`Service Name` field.

```

File: content/applications/inventory_and_mrp/inventory/shipping_receiving/setup_configuration/starshipit_shipping.rst
Comment (Felicious) [inline content/applications/inventory_and_mrp/inventory/shipping_receiving/setup_configuration/starshipit_shipping.rst:134]: What is Sendle? Is it a well-known delivery service?

File: content/applications/inventory_and_mrp/inventory/shipping_receiving/setup_configuration/starshipit_shipping.rst
Comment (Felicious) [inline content/applications/inventory_and_mrp/inventory/shipping_receiving/setup_configuration/starshipit_shipping.rst:151]:
```suggestion
   immediately canceled after creation — this occurs automatically. Please note that depending on
   the shipping provider being used, charges may still occur and you may need to cancel your orders
   on the provider platform.
   
   Switch between the test and production environment by clicking the :guilabel:`Environment` smart
   button at the top of the shipping method form.
```

File: content/applications/inventory_and_mrp/inventory/shipping_receiving/setup_configuration/starshipit_shipping.rst
Comment (Felicious) [inline content/applications/inventory_and_mrp/inventory/shipping_receiving/setup_configuration/starshipit_shipping.rst:158]:
```suggestion
When creating a quotation in Odoo, add the Starshipit shipping method by clicking the :guilabel:`Add
shipping` button.

In the :guilabel:`Add a shipping method` pop-up window, select Starshipit in the :guilabel:`Shipping
Method` field.

Calculate the shipping rate by clicking :guilabel:`Get rate`.

Finally, click :guilabel:`Add` to include the cost of shipping to the sales order line, labeled as
the *delivery product*.

.. note::
   Automatically calculate shipping costs for Starshipit in **both** Odoo *Sales* and *eCommerce*
   applications.
```

**Review by Felicious — APPROVED** [review]
Hi @vin-odoo ! I just tested the Starshipit module, and everything works as documented! Incredible work with this highly detailed and accurate doc! 

After you go through my comments, feel free to move this document onto the next stage of review. Please let me team lead @StraubCreative know if you need assistance merging this document to the 17.0 documentation branch.

File: content/applications/inventory_and_mrp/inventory/shipping_receiving/setup_configuration/starshipit_shipping.rst
Comment (Felicious) [reply content/applications/inventory_and_mrp/inventory/shipping_receiving/setup_configuration/starshipit_shipping.rst:86]: Thank you so much for getting back to me so quickly!

Comment (vin-odoo) [reply content/applications/inventory_and_mrp/inventory/shipping_receiving/setup_configuration/starshipit_shipping.rst:86]: @Felicious Thank you for the review! I will be working with our Australian colleagues during the day to address your comments and most likely ask for another review afterward :pray:

File: content/applications/inventory_and_mrp/inventory/shipping_receiving/setup_configuration/starshipit_shipping.rst
Comment (vin-odoo) [reply content/applications/inventory_and_mrp/inventory/shipping_receiving/setup_configuration/starshipit_shipping.rst:134]: Just one of the shipping carrier in Australia that is available in Starshipit.
As it was the easiest to set up for testing we used it a lot, so I just used it as an example.

**Pull request closed without merging**
